app/src/test/java/org/grammatek/simacorrect/spellcheckerservice/YfirlesturAnnotationTest.kt

```diff
@@ -0,0 +1,219 @@
+        val suggestionLimit = 5
+        val dictionary = arrayListOf<String>()
+        val texts = listOf(
+            "Zwart",
+            "Zwart er",
+            "Zwart er vedrid",
+            "Zwart er vedrid i",
+            "Zwart er vedrid i i",
+            "Zwart er vedrid i i dag"
+        )
+        val expectedAnnotations = listOf(
+            listOf(),
+            listOf(),
+            listOf(AnnotationIndices(0, 14)),
+            listOf(AnnotationIndices(9, 14), AnnotationIndices(16, 16)),
+            listOf(AnnotationIndices(9, 14), AnnotationIndices(16, 18)),
+            listOf(AnnotationIndices(9, 14), AnnotationIndices(16, 18)),
+        )
+
+        for (i in texts.indices) {
+            val response = api.correctApiPost(texts[i])
+            val ylAnnotations = YfirlesturAnnotation(response, texts[i])
+            ylAnnotations.getSuggestionsForAnnotatedWords(suggestionLimit, dictionary)
+            val annotations = ylAnnotations.suggestionsIndices
+
+            assertThat(annotations.size).isEqualTo(expectedAnnotations[i].size)
+            for (j in annotations.indices) {
+                assertThat(annotations[j].startChar).isEqualTo(expectedAnnotations[i][j].startChar)
+                assertThat(annotations[j].endChar).isEqualTo(expectedAnnotations[i][j].endChar)
+            }
+        }
+    }
+
+    @Test
+    fun `does Yfirlestur token count match our constructed token count`() {
+        val suggestionLimit = 5
+        val dictionary = arrayListOf<String>()
+        for (testCase in testCases) {
+            val response = api.correctApiPost(testCase)
+            val ylAnnotation = YfirlesturAnnotation(response, testCase)
+            ylAnnotation.getSuggestionsForAnnotatedWords(suggestionLimit, dictionary)
+            var size = 0
+            for (tokensIndex in ylAnnotation.tokensIndices) {
+                size += tokensIndex.size
+            }
+            assertThat(size).isEqualTo(response.stats?.numTokens ?: 0)
+        }
+    }
+
+    @Test
+    fun `is dictionary taken into account for suggestions`() {
+        val suggestionLimit = 5
+        val text = "Eg dreymi"
+        val response = api.correctApiPost(text)
+        val dictionary = arrayListOf<String>()
+        var suggestionsInfo = YfirlesturAnnotation(response, text).getSuggestionsForAnnotatedWords(suggestionLimit, dictionary)
+        assertThat(suggestionsInfo).isNotEmpty()
+
+        dictionary.add("Eg")
+        suggestionsInfo = YfirlesturAnnotation(response, text).getSuggestionsForAnnotatedWords(suggestionLimit, dictionary)
+        assertThat(suggestionsInfo).isEmpty()
+
+        dictionary.clear()
+        suggestionsInfo = YfirlesturAnnotation(response, text).getSuggestionsForAnnotatedWords(suggestionLimit, dictionary)
+        assertThat(suggestionsInfo).isNotEmpty()
+    }
+
+    @Test
+    fun `is suggestion limit enforced`() {
+        var suggestionLimit = 0
```

Comment:
this test would benefit from much more raw text variations and increasing suggestionLimits from 0..n

Reply:
The only problem with that is it's hard to get more than 2 suggestions from Yfirlestur for a single annotation. Realistically Yfirlestur only gives 1 Annotation per word.